### PR TITLE
Fixing twitter profile image blank issue

### PR DESCRIPTION
**Pull request by thiago-preira**
Withheld tweets does not contain a proper profile image url, as API response extracted below:
``` JSON
{
  "created_at":"Thu Mar 03 09:53:12 +0000 2022",
  "id":1499321986073432067,
  "id_str":"1499321986073432067",
  "text":"@sputnik_ar's account has been withheld in Belgium, Austria, Bulgaria, Sweden, Croatia, Spain, Slovenia, Cyprus, Slovakia, Czech Republic, Romania, Portugal, Poland, Denmark, Netherlands, Estonia, Malta, Luxembourg, Finland, France, Lithuania, Germany, Greece, Latvia, Hungary, Italy, Ireland in response to a legal demand. Learn more.",
...
"profile_image_url":"",
"profile_image_url_https":"",
...
}

```
This causes an `java.lang.StringIndexOutOfBoundsException: String index out of range: -1` at `ProfileImage.scala` line 18
To fix the issue I'm just forwarding the empty url for images

### CODE REVIEW

**Comment by thiago-preira** [comment]
Hi Daniela, thanks for your reply.
As mentioned above, the twitter API sends the `profile_image_url` and `profile_image_url_https` fields as empty string rather than null, that's why I have implemented the fix that way. 
Please let me know if there is anything that you think it should be updated.
Cheers,
Thiago

**Comment by DanielaSfregola** [comment]
Hi @thiago-preira,
we fully control how we serialize/deserialize from Twitter, so I propose we do the following:

```
private[twitter4s] case object ProfileImageSerializer
    extends CustomSerializer[ProfileImage](format =>
      ({
        case JString("") => null // we consider returning an empty string equivalent to null
        case JString(n) => ProfileImage(n)
        case JNull      => null
      }, {
        case img: ProfileImage => JString(img.normal)
      }))
      
```
from CustomFormats.scala

And then we change `ProfileImage` to be an `Option[ProfileImage]` in the case classes where this is used.
This is indeed an API change (we'll need a version bump!), but it better represents that a profile image is not available for the user.

WDYT?

Cheers,
Daniela

**Comment by thiago-preira** [comment]
Makes total sense! Sorry, wasn't aware of custom serialization/deserialization. Definitely way more elegant solution.
Not sure about the version bump. I have increased to 8.0
I have applied the fixes, please let me know if any other adjustments are needed.
Cheers,
Thiago

File: src/main/scala/com/danielasfregola/twitter4s/http/serializers/CustomFormats.scala [hunk 1]
```diff
@@ -65,8 +65,9 @@ private[twitter4s] case object DisconnectionCodeSerializer
 private[twitter4s] case object ProfileImageSerializer
     extends CustomSerializer[ProfileImage](format =>
       ({
-        case JString(n) => ProfileImage(n)
-        case JNull      => null
+        case JString("") => null
```
Comment (DanielaSfregola) [inline src/main/scala/com/danielasfregola/twitter4s/http/serializers/CustomFormats.scala:68]: could you add a comment here to remind why we do it?

File: src/main/scala/com/danielasfregola/twitter4s/entities/User.scala [hunk 2]
```diff
@@ -33,8 +33,8 @@ final case class User(blocked_by: Boolean = false,
                       profile_background_image_url_https: String,
                       profile_background_tile: Boolean = false,
                       profile_banner_url: Option[String] = None,
-                      profile_image_url: ProfileImage,
-                      profile_image_url_https: ProfileImage,
+                      profile_image_url: Option[ProfileImage] = None,
```
Comment (DanielaSfregola) [inline src/main/scala/com/danielasfregola/twitter4s/entities/User.scala:36]: Why the defaults? I don't think we need them

File: build.sbt [hunk 3]
```diff
@@ -3,7 +3,7 @@ import com.typesafe.sbt.SbtGit.{GitKeys => git}
 enablePlugins(GhpagesPlugin, SiteScaladocPlugin)
 
 name := "twitter4s"
-version := "7.2-SNAPSHOT"
+version := "8.0"
```
Comment (DanielaSfregola) [inline build.sbt:6]: Please revert this change, I'll make sure to take care of it when doing a release :)

**Review by DanielaSfregola — CHANGES_REQUESTED** [review]
Hi @thiago-preira,
no worries! That is exactly why we do code reviews :)

I have added a few comment, please address them and we should be read to merge.

Also, it would be great if we could add a test for this -- you could just change one of the existing mock jsons and set a profile image to "".

Cheers,
Daniela

File: build.sbt (hunk 3 above)
Comment (thiago-preira) [reply build.sbt]: reverted

File: src/main/scala/com/danielasfregola/twitter4s/entities/User.scala (hunk 2 above)
Comment (thiago-preira) [reply src/main/scala/com/danielasfregola/twitter4s/entities/User.scala]: removed

File: src/main/scala/com/danielasfregola/twitter4s/http/serializers/CustomFormats.scala (hunk 1 above)
Comment (thiago-preira) [reply src/main/scala/com/danielasfregola/twitter4s/http/serializers/CustomFormats.scala]: comment added

**Review by DanielaSfregola — APPROVED** [review]
Hi @thiago-preira ,
thanks for your contribution!

LGTM

**Pull request merged**